src/modules/admin/components/formRules/NewFormRuleDialog.tsx

```diff
@@ -128,6 +130,25 @@ const NewFormRuleDialog: React.FC<Props> = ({
     },
   });
 
+  // This would be better handled with real form validation
+  const [validationError, setValidationError] = useState<string>();
+  const handleSubmit = () => {
+    let error: string | undefined = undefined;
+    if (formRole === FormRole.Service) {
+      if (rule.serviceCategoryId && rule.serviceTypeId) {
```

Comment:
I'm reading the "type OR category" as "XOR" but perhaps we have some case where someone would set both?

Reply:
Yes I agree with you that it is xor.

When we wrote the ticket I think we were envisioning that this would be another horizontal pair of dropdowns, similar to the Conditions you can add below. In the first dropdown you would select Service Type or Service Category, and in the second dropdown you would pick from a dynamic pick list whose contents change depending on the value of the first dropdown. 

That would eliminate the need for an error validation step since the interface would inherently prevent the user from picking both service type and service category.

I think it would be nice to use this design consistently, but I also agree with your concern about the accessibility of this pattern, so let's talk about it with design if we can't figure out a more accessible implementation.